web/blog/parquet-and-feather-writing-security-telemetry/index.qmd

```diff
@@ -0,0 +1,868 @@
+---
+title: "Parquet & Feather: Writing Security Telemetry"
+authors:
+  - dispanser
+  - mavam
+date: 2022-10-22
+image: image.jpg
+tags: [benchmark, arrow, parquet, feather, quarto, r]
+---
+
+How does Apache [Parquet][parquet] compare to [Feather][feather] for storing
+structured security data? In this blog post, we answer this question.
+
+[parquet]: https://parquet.apache.org/
+[feather]: https://arrow.apache.org/docs/python/feather.html
+
+<!--truncate-->
+
+:::info Parquet & Feather: 2/3
+This is blog post is part of a 3-piece series on Parquet and Feather.
+
+1. [Enabling Open Investigations][parquet-and-feather-1]
+2. Writing Security Telemetry (this blog post)
+3. TBD
+
+[parquet-and-feather-1]: /blog/parquet-and-feather-enabling-open-investigations/
+:::
+
+In the [previous blog][parquet-and-feather-1], we explained why Parquet and
+Feather are great building blocks for modern investigations. In this blog, we take
+a look at how they actually perform on the write path in two dimensions:
+
+- **Size**: how much space does typical security telemetry occupy?
+- **Speed**: how fast can we write out to a store?
+
+Parquet and Feather have different goals. While Parquet is an on-disk format
+that optimizes for size. Feather is a thin layer around the native Arrow
+in-memory representation. This puts them at different points in the spectrum of
+throughput and latency.
+
+To better understand this spectrum, we instrumented the write path of VAST,
+which consists roughly of the following steps:
+
+1. Parse the input
+2. Convert it into Arrow record batches
+3. Ship Arrow record batches to a VAST server
+4. Write Arrow out into a Parquet or Feather store
+
+Since steps (1–3) are the same for both stores, we ignore them in the following
+analysis and solely zoom in on (4).
```

Comment:
```suggestion
4. Write Arrow record batches out into a Parquet or Feather store
5. Create an index from Arrow record batches

Since steps (1–3) and (5) are the same for both stores, we ignore them in the following
analysis and solely zoom in on (4).
```